definitions/checks/foreman/check_puppet_capsules.rb

```diff
@@ -0,0 +1,43 @@
+module Checks
+  class CheckPuppetCapsules < ForemanMaintain::Check
+    metadata do
+      label :puppet_capsules
+      for_feature :foreman_database
+      description 'Check for Puppet capsules from the database'
+      manual_detection
+    end
+
+    def run
+      puppet_proxies = find_puppet_proxies.reject { |proxy| local_proxy?(proxy['url']) }
+      unless puppet_proxies.empty?
+        names = puppet_proxies.map { |proxy| proxy['name'] }
+        print('You have proxies with Puppet feature enabled, '\
+              "please disable Puppet on all proxies first.\n"\
+              "The following proxies have Puppet feature: #{names.join(', ')}.")
+        abort!
+      end
+    end
+
+    def find_puppet_proxies
+      feature(:foreman_database).query(puppet_proxies_query)
+    end
+
+    private
+
+    def local_proxy?(url)
+      URI.parse(url).hostname.casecmp(hostname) == 0
```

Comment:
Isn't this the same?
```suggestion
      URI.parse(url).hostname.casecmp?(hostname)
```

Reply:
It is apart of the version support... The `casecmp?` is not part of 2.0 apparently